src/nvim/eval/userfunc.c

```diff
@@ -162,6 +163,17 @@ static void register_closure(ufunc_T *fp)
     [current_funccal->fc_funcs.ga_len++] = fp;
 }
 
+
+/// Get a name for a lambda.  Returned in static memory.
+char_u * get_lambda_name(void)
+{
+    static char_u   name[30];
```

Comment:
This isn't thread safe.  I know right now we only have the main and tui threads, so there _shouldn't_ be a problem but it seems cleaner to just have the caller provide the buffer.

Reply:
I just ported this over from vim's code -- they added this in a later commit. I didn't patch the whole thing though.

Reply:
Oh, I think I see what you're saying. Instead of taking `void`, I should just pass in `char_u *` pointer and let it set there, so there's no problem with getting the wrong buffer or some other dumb race condition.

Reply:
Hmmm, I don't think I'll actually do this in this PR, since I've just ported the vim implementation for this if that's OK. Do we have some note to leave for not thread-safe stuff?

Reply:
I also don't think this makes sense. It seems safe in its usage, but it also seems like a rather unnecessary change.